### PR TITLE
Add test case cli

### PR DESCRIPTION
**Pull request by linjunjj**
Add test case cli ， include client cli , logger cli, trace cli, vm cli etc.

### CODE REVIEW

File: test/emq_mgmt_SUITE.erl [hunk 1]
```diff
@@ -64,3 +84,161 @@ t_check_app(_Config) ->
...
+
+t_broker_cmd(_) ->
+    ct:pal("start test  case broker cli"),
+    emqx_mgmt_cli:broker(["stats"]),
+    emqx_mgmt_cli:broker(["metrics"]).
+
+t_clients_cmd(_) ->
+    ct:pal("start test case client cli"),
+    process_flag(trap_exit, true),
+    {ok, T1} = emqx_client:start_link([{host, "localhost"},
```
Comment (tigercl) [inline test/emq_mgmt_SUITE.erl]: Align the code

File: test/emq_mgmt_SUITE.erl [hunk 2]
```diff
@@ -64,3 +84,157 @@ t_check_app(_Config) ->
...
+
+t_subscriptions_cmd(_) ->
+    ct:pal("start test subscriptions cli"),
+    {ok, T4} = emqx_client:start_link([{host, "localhost"},
+                                       {client_id, <<"client4">>},
+                                       {username, <<"testuser4">>},
+                                       {password, <<"pass4">>}]),
+    emqx_client:connect(T4),
+    emqx_client:subscribe(T4, <<"/b/b/c">>),
+    ?assertMatch({match, _}, re:run(emqx_mgmt_cli:subscriptions(["add", "client4", "/b/b/c", "0"]), "ok")),
```
Comment (Gilbert-Wong) [inline test/emq_mgmt_SUITE.erl]: `?assertEqual(["ok~n"], emqx_mgmt_cli:subscriptions(["add", "client4", "/b/b/c", "0"]))`. Is it necessary to use `regular expressions` in your code?

File: src/emqx_mgmt_cli.erl [hunk 3]
```diff
@@ -122,12 +129,12 @@ mgmt(_) ->
 
 status([]) ->
     {InternalStatus, _ProvidedStatus} = init:get_status(),
-    emqx_cli:print("Node ~p is ~p~n", [node(), InternalStatus]),
+        ?PRINT("Node ~p is ~p~n", [node(), InternalStatus]),
     case lists:keysearch(?APP, 1, application:which_applications()) of
         false ->
-            emqx_cli:print("~s is not running~n", [?APP]);
+            ?PRINT("~s is not running~n", [?APP]);
```
Comment (Gilbert-Wong) [inline src/emqx_mgmt_cli.erl]: Using emqx_cli and assert these results. Stop using regular expression match.

File: src/emqx_mgmt_cli.erl [hunk 4]
```diff
@@ -17,9 +17,17 @@
...
+-define(PRINT_MSG(Msg), io_lib:format("~s", [Msg])).
+-else.
 -define(PRINT_MSG(Msg), io:format(Msg)).
+-endif.
 
+-ifdef(TEST).
+-define(PRINT(Format, Args), io_lib:format(Format, Args)).
+-else.
 -define(PRINT(Format, Args), io:format(Format, Args)).
+-endif.
```
Comment (terry-xiaoyu) [inline src/emqx_mgmt_cli.erl]: Put PRINT_MSG/1 and PRINT_MSG/2 into one -ifdef(TEST) clause:
```erlang
-ifdef(TEST).
-define(PRINT_MSG(Msg), io_lib:format("~s", [Msg])).
-define(PRINT(Format, Args), io_lib:format(Format, Args)).
-else.
-define(PRINT_MSG(Msg), emqx_cli:print(Msg)).
-define(PRINT(Format, Args), emqx_cli:print(Format, Args)).
-endif.
```

File: src/emqx_mgmt_cli.erl [hunk 5]
```diff
@@ -17,9 +17,17 @@
 -include_lib("emqx/include/emqx.hrl").
 -include_lib("emqx/include/emqx_mqtt.hrl").
 
+-ifdef(TEST).
+-define(PRINT_MSG(Msg), io_lib:format("~s", [Msg])).
+-else.
 -define(PRINT_MSG(Msg), io:format(Msg)).
```
Comment (terry-xiaoyu) [inline src/emqx_mgmt_cli.erl]: Use `emqx_cli:print/1` instead.

File: src/emqx_mgmt_cli.erl [hunk 6]
```diff
@@ -17,9 +17,17 @@
...
+-endif.
 
+-ifdef(TEST).
+-define(PRINT(Format, Args), io_lib:format(Format, Args)).
+-else.
 -define(PRINT(Format, Args), io:format(Format, Args)).
```
Comment (terry-xiaoyu) [inline src/emqx_mgmt_cli.erl]: Use `emqx_cli:print/2` instead.

File: test/emq_mgmt_SUITE.erl [hunk 7]
```diff
@@ -64,3 +84,161 @@ t_check_app(_Config) ->
...
+    ?assertMatch({match, _}, re:run(emqx_mgmt_cli:mgmt(["delete", "emqx_appid"]), "ok")).
+
+t_status_cmd(_) ->
+    ct:pal("start testing status command"),
+    emqx_mgmt_cli:status([]).
+
+t_broker_cmd(_) ->
+    ct:pal("start testing  the broker command"),
+    ?assertMatch({match, _}, re:run(emqx_mgmt_cli:broker(["stats"]), "_")),
+    ?assertMatch({match, _}, re:run(emqx_mgmt_cli:broker(["metrice"]), "_")).
```
Comment (terry-xiaoyu) [inline test/emq_mgmt_SUITE.erl]: what would '_' be matched?

File: test/emq_mgmt_SUITE.erl [hunk 8]
```diff
@@ -64,3 +82,163 @@ t_check_app(_Config) ->
...
+    ?assertMatch({match, _}, re:run(emqx_mgmt_cli:mgmt(["delete", "emqx_appid"]), "ok")).
+
+t_status_cmd(_) ->
+    ct:pal("start testing status command"),
+    ?assertMatch({match, _}, re:run(emqx_mgmt_cli:status([]), "is running")).
+
+t_broker_cmd(_) ->
+    ct:pal("start testing the broker command"),
+    ?assertMatch({match, _}, re:run(emqx_mgmt_cli:broker(["stats"]), "_")),
+    ?assertMatch({match, _}, re:run(emqx_mgmt_cli:broker(["metrics"]), "_")).
```
Comment (terry-xiaoyu) [inline test/emq_mgmt_SUITE.erl]: Are these commands expected "_" outputs?

**Comment by terry-xiaoyu** [comment]
Close this as we use https://github.com/emqx/emqx-management/pull/24 to merge the PR.

**Pull request closed without merging**